kafka-1.0.x/src/main/scala/monix/kafka/CommittableOffsetBatch.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2014-2019 by The Monix Project Developers.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package monix.kafka
+
+import monix.eval.Task
+import org.apache.kafka.common.TopicPartition
+
+/** Batch of Kafka offsets which can be committed together.
+  * Can be builded from offsets sequence by [[CommittableOffsetBatch#apply]] method.
+  * Besides you can use [[CommittableOffsetBatch#empty]] method to create empty batch and
+  * add offsets to it by [[updated]] method.
+  *
+  * Anyway offsets order make sense! Only last added offset for topic and partition will
+  * be committed to Kafka.
+  *
+  * @param offsets is the offsets batch for a few topics and partitions.
+  *        Make sure that each of them was received from one [[KafkaConsumerObservable]].
+  *
+  * @param commitBatch is the function for batched commit realized as closure
+  *        in [[KafkaConsumerObservable]] context. This decision was made for
+  *        thread-safety reasons. This parameter is obtained from last [[CommittableOffset]]
+  *        added to batch.
+  */
+final class CommittableOffsetBatch(
+  val offsets: Map[TopicPartition, Long],
+  commitBatch: Map[TopicPartition, Long] => Task[Unit]) {
```

Comment:
Should this also be private?

Reply:
It is private already. `CommitableOffsetBatch` is a class, not case class. Therefore constructor params not marked as `val` is not available outside:

```
@ final class Sample(val available: Int, notAvailable: Int)
defined class Sample

@ new Sample(1, 2)
res1: Sample = ammonite.$sess.cmd0$Sample@676f0a60

@ res1.available
res2: Int = 1

@ res1.notAvailable
cmd3.sc:1: value notAvailable is not a member of ammonite.$sess.cmd0.Sample
val res3 = res1.notAvailable
                ^
Compilation Failed
```

Reply:
Right, my bad!